### ISSUE TITLE
Inverted error message for no-descending-specificity rule?

### ISSUE THREAD

**Issue by ryanheathers** [stylelint/stylelint#1678]
Possible bug with the warning message. I've enabled the no-descending-specificity rule and after running it, I'm getting the following warning: 

`Expected selector ".container-overlay a:-webkit-any-link" to come before selector ".container-share-links a" (no-descending-specificity)`.

My CSS:

**Line 27**
```
.container-overlay a:-webkit-any-link {
  text-decoration: none;
}
```

**Line 238**

```
.container-share-links a {
  text-decoration: none;
}
```

The rule is correctly catching a specificity issue, however the warning message is inverted, right? The more specific selector should come **after** the more general one. Yet the warning message is advising the opposite, unless I'm getting my brain twisted into knots.

I'm using `version 7.0.3` of stylelint. Thanks!

**Comment by jeddy3** [comment]
@ryanheathers Thanks for the report. 

Yes, it appears to be back to front. I'm surprised this has only be reported now as the rule is a few months old.

A simpler tests shows it's backwards:

```
ok 16 should be accepted
#
> rule: no-descending-specificity
> config: true
> code: "b a {} a {}"

ok 17 should register one warning
ok 18 should warn on line 1
ok 19 should warn on column 8
ok 20 should warn with message Expected selector "b a" to come before selector "a" (no-descending-specificity)
```

I've created a PR to fix it, which switches things around:

```
ok 16 should be accepted
#
> rule: no-descending-specificity
> config: true
> code: "b a {} a {}"

ok 17 should register one warning
ok 18 should warn on line 1
ok 19 should warn on column 8
ok 20 should warn with message Expected selector "a" to come before selector "b a" (no-descending-specificity)
```
